index.html

```diff
@@ -923,7 +923,7 @@ <h4>Text</h4>
           <p>A <a>Text</a> object is represented in a <a>DAPT Document</a> by a <code>&lt;p&gt;</code> element with the following constraints:</p>
             <ul>
               <li>The <a>Text</a> of the <a>Script Event</a> is represented by the character content
-              of the <code>&lt;p&gt;</code> and its descendants, and after applying White Space Handling
+              of the <code>&lt;p&gt;</code> and of its descendants, and after applying White Space Handling
```

Comment:
Really sorry, I don't know why I'm thinking of these issues one at a time rather than all together, but there's another issue: not _all_ of its descendants' character content is the Text - for example any descendant of a `<metadata>` child would not count.

I'm not sure quite how to word it, because `<span>`s can have nested `<span>` children themselves, and so on, but it would also be possible (with undefined semantics) to have a `<span>` descendant of a `<metadata>` element or some other foreign namespace element.

There is probably a better way, but here's what I came up with:

```suggestion
              of the <code>&lt;p&gt;</code> element
              and of its <code>&lt;span&gt;</code> element children
              and (recursively) their <code>&lt;span&gt;</code> element children,
              and after applying White Space Handling
```

Reply:
What about `<br>` elements?

Reply:
Hmm, how about inserting before ", and after applying White Space Handling" an additional line:

>, where `&lt;br&gt;` elements are considered to represent new lines within the character content

?

Reply:
What about ruby annotation text? How should we consider it? Should we be silent about it?

Reply:
I think we can be silent about it because ruby annotation text is part of the Text, and the mechanism for inserting it uses nested `<span>`s whose content is already included in the definition. Unless your view is that ruby annotations are _not_ part of the Text, @cconcolato ?

Reply:
Let's stay silent about it. I pushed the change for line breaks.

Reply:
I try to address your comments (also squashed and rebased against the main branch to resolve conflicts)

Reply:
Is that 0512851 to resolve https://github.com/w3c/dapt/pull/158#discussion_r1455055567 ? I don't think that wording rejects character content children of  `<span>` elements inside foreign namespace elements - I think we agree that such content should not be included in the definition of Text?

Reply:
I added pruning (but it probably should be rephrased even more, as pruning should happen first). 

I'm not convinced we need all of that text. So far, the spec is written using the following structure: 
"The [part of the DAPT model] is represented by [some syntax in TTML]"
I think it is expressing a one way mapping: how from the model you build the syntax, not how from the syntax you obtain the model. 

Maybe we want that bidirectional mapping? That would require reviewing the rest of the spec (e.g. what if the TTML elements representing a Character, contain more than the few elements specified in the spec).

Assuming, the mapping is one way for now (for consistency with other parts of the spec), the text in this PR should only say that the Text object is represented by a p element, possibly structured in spans, possibly with timing, possibly with interspersed metadata and foreign elements. All the "possibly" parts are covered by notes already.

Reply:
@nigelmegitt I would be interested in your feedback on the above comment.

Reply:
I'm still thinking about this!

My instinct is that it's fine as is, and maybe we should open a separate issue about this question. I think the key question for implementers is what they should do if they encounter a TTML2 document that claims to be a DAPT document but doesn't exactly match the data model. Are we telling them to implement all relevant TTML2 semantics or only those specifically mentioned in DAPT? Formally I believe it is the set of TTML2 features defined as being required for processors. From that perspective I don't believe we need to say any more about it. 

But happy to discuss further.